### PR TITLE
EWL-3793 Topics | Style Hero

### PR DESCRIPTION
**Pull request by chelsiejohnston**
## Ticket(s)

**Jira Ticket**

- [EWL-3793: Topics | Style Hero](https://issues.ama-assn.org/browse/EWL-3793)
- [EWL-3808: Topics | Style Hero](https://issues.ama-assn.org/browse/EWL-3808)

### Suggestion for how to organize patterns:

img_7093

**(A)** Atoms are created. These are all pretty much **base patterns** and are comprised of a single element.

**(B)** Atoms are combined to create a base Molecule. In this PR, `hero-article-preview` is the base Molecule pattern. **Molecules are the main type of pattern to receive a variant**.

**(C)** Evaluate whether the new pattern requirement deviates from a existing base molecule pattern. For example: is its responsive behavior the same? Is its requirement for column widths at different breakpoints the same? Is the markup totally different, or is the new pattern otherwise different enough from a base molecule that it can't be controlled by passing a class? We may be able to define better or more specific parameters here.

**(D)** If the new molecule is different enough that it can't easily be reused (for example by passing a CSS class), it should be a variant. Variants should be stored _within_ the base Molecule pattern's directory:

screen shot 2017-07-25 at 4 49 56 pm
^ Example nesting of variant `hero-article-preview-video` inside the base Molecule pattern directory `hero-article-preview`. 

**(E)** Template patterns that require variant Molecule patterns should have Template-specific Organism patterns that are not assumed to be reusable. In this PR, the Template-specific Organism pattern is `topic-hero-section`. We can tell it's specific to the`topic` Template pattern from its namespacing. Template-specific Organisms may be organized by Template within the Organisms directory so that all of the Organisms are in the same place.

If one of these patterns appears more than once in our requirements, it could become a new base pattern, but the main takeaways are: 

1) Molecules are reusable, and are the main type of pattern to get variants
2) Organisms are not necessarily reusable, are specific to the needs of the Template, and contain base and variant Molecules.

## To Test

- [x] Templates > Topic > Topic - compare with wires / code review / visual QA
- [x] Organisms > Topic > Topic Hero Section - compare with wires / code review / visual QA
- [x] Molecules > Components > Hero (Article Preview) - compare with wires / code review / visual QA
- [x] Molecules > Components > Hero (Article Preview - Video variant) - compare with wires / code review / visual QA

## Relevant Screenshots/GIFs

screen shot 2017-07-25 at 5 15 37 pm

## Future Tasks

- [x] I added widths to the layout columns in topic.twig to start getting a basic sense of how the Hero section will look, but these should be taken as placeholders since they're just percentages currently.
- [x] The `@inprogress` pattern state seems to not be working properly. A bunch of patterns are highlighted with that state that shouldn't be, FYI.
- [x] Right now there are inline styles for demo purposes, and also because I haven't added any custom .scss to the new patterns yet. I can remove the demo things before this gets merged, and we can do this as part of the next round of visual QA.
- [ ] [Hiding variant patterns](http://patternlab.io/docs/pattern-hiding.html) might be a good idea to avoid clogging up the list
- [ ] The `` in "Hero (Article Preview)" is not atomized yet. The existing atom was too small since we set it to dimensions for the EC. Not sure how to proceed here

## Additional Notes

Very rough, but I wanted to get your guys' thoughts on the organization system quickly!

### CODE REVIEW

**Comment by rpama** [comment]
@deadarm found this issue reported, indicating that Pattern States seem to be broken. So looks like we should remove the @inprogress flag.
https://github.com/pattern-lab/patternlab-php/issues/386

**Comment by deadarm** [comment]
- Agreed, we should not attempt to use states at this time due to a know bug.
- I think @chelsiejohnston's outline of how to organize makes a lot of sense to me.
- I am seeing two topic directories in the templates directory, so not sure if that is needed or if 05-topic can be removed?
- I think the presented way forward with regard to molecules and molecule variants is reasonable. I would even hope that if we are presented with an item that could almost be made a variant of something, but there are one or two things we think COULD justify making a new pattern, that we would push to have it be a variant instead of a new pattern in order to maintain a sensible catalog of patterns.

**Comment by rpama** [comment]
I'm in agreement with all of the above. To summarize what I see as the verdict here, the two primary takeaways I'm seeing are:
1) Don't give molecules template-specific names, but do create custom variants as needed and nest them in the base molecule directory.
2) Organisms can be specific to a template.

I would like to further recommend that each variant get it's own scss file, or as @deadarm put it "if there is a twig template, there can be an scss sheet, that only has stuff under the variant class."

Assuming the above is correct, I think we have a way to proceed with creating the rest of our Topic patterns, and this PR just needs to be cleaned up (instructional boxes removed, patternstate removed, etc) and then we'll re-review. Sound good @chelsiejohnston?

**Comment by chelsiejohnston** [comment]
I added styling from the UX meeting since I had the time. 

We should reuse some of this styling in other patterns (the article metadata, etc.) but I left it in the pattern-level scss for now. 

Ready for re-review.

File: styleguide/source/_patterns/03-templates/04-topic/topic.twig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<header>
+  {% include 'organisms-ribbon' %}
+</header>
+
+<main class="topic" style="display: flex; flex-direction: row;">
```
Comment (deadarm) [inline styleguide/source/_patterns/03-templates/04-topic/topic.twig]: Suggestion to remove or move inline CSS in this file into a new Scss sheet in the Topic directory.

**Comment by chelsiejohnston** [comment]
@deadarm I'm resolving the merge conflicts now, and then I'll merge it if you're cool with me self-merging :)

**Pull request merged**